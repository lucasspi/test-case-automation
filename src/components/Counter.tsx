import React, { useState } from 'react';

interface CounterProps {
  initialValue?: number;
  step?: number;
}

export default function Counter({ initialValue = 0, step = 1 }: CounterProps) {
  const [count, setCount] = useState(initialValue);

  const increment = () => setCount(prev => prev + step);
  const decrement = () => setCount(prev => prev - step);
  const reset = () => setCount(initialValue);

  return (
    <div className="counter">
      <h2>Counter: {count}</h2>
      <div className="buttons">
        <button onClick={decrement}>-</button>
        <button onClick={increment}>+</button>
        <button onClick={reset}>Reset</button>
      </div>
    </div>
  );
}
