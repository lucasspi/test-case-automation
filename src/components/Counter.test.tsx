import { render, screen, fireEvent } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Counter from './Counter';

describe('Counter', () => {
  it('renders without crashing', () => {
    render(<Counter />);
    expect(screen.getByText(/Counter:/)).toBeInTheDocument();
  });

  it('renders with default initial value', () => {
    render(<Counter />);
    expect(screen.getByText('Counter: 0')).toBeInTheDocument();
  });

  it('renders with custom initial value', () => {
    render(<Counter initialValue={5} />);
    expect(screen.getByText('Counter: 5')).toBeInTheDocument();
  });

  it('increments counter when + button is clicked', () => {
    render(<Counter />);
    const incrementButton = screen.getByText('+');
    
    fireEvent.click(incrementButton);
    expect(screen.getByText('Counter: 1')).toBeInTheDocument();
    
    fireEvent.click(incrementButton);
    expect(screen.getByText('Counter: 2')).toBeInTheDocument();
  });

  it('decrements counter when - button is clicked', () => {
    render(<Counter initialValue={5} />);
    const decrementButton = screen.getByText('-');
    
    fireEvent.click(decrementButton);
    expect(screen.getByText('Counter: 4')).toBeInTheDocument();
  });

  it('uses custom step value', () => {
    render(<Counter initialValue={0} step={5} />);
    const incrementButton = screen.getByText('+');
    
    fireEvent.click(incrementButton);
    expect(screen.getByText('Counter: 5')).toBeInTheDocument();
  });

  it('resets counter to initial value when reset button is clicked', () => {
    render(<Counter initialValue={10} />);
    const incrementButton = screen.getByText('+');
    const resetButton = screen.getByText('Reset');
    
    // Increment a few times
    fireEvent.click(incrementButton);
    fireEvent.click(incrementButton);
    expect(screen.getByText('Counter: 12')).toBeInTheDocument();
    
    // Reset
    fireEvent.click(resetButton);
    expect(screen.getByText('Counter: 10')).toBeInTheDocument();
  });

  it('has all required buttons', () => {
    render(<Counter />);
    
    expect(screen.getByText('+')).toBeInTheDocument();
    expect(screen.getByText('-')).toBeInTheDocument();
    expect(screen.getByText('Reset')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Counter initialValue={5} step={2} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});