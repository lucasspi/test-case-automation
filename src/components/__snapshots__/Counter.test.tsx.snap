// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > matches snapshot 1`] = `
<div
  class="counter"
>
  <h2>
    Counter: 
    5
  </h2>
  <div
    class="buttons"
  >
    <button>
      -
    </button>
    <button>
      +
    </button>
    <button>
      Reset
    </button>
  </div>
</div>
`;
